### PR TITLE
Add ability to remove track from playlist on mobile

### PR DESCRIPTION
**Pull request by jowlee**
### Description
Adds the ability to remove track from playlist on mobile. 
Created a Drawer component to confirm delete track. 
Since the mobile edit playlist stores changes locally and only commits them when the user hit save, 
the deleted tracks are stored in an array and all deleted at the end. 

Should I move this icon to stems & bump and import? 
Closes AUD-46

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
nope

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.
I ran this on the ios simulator to test.

### CODE REVIEW

File: src/containers/edit-playlist/mobile/EditPlaylistPage.tsx [hunk 1]
```diff
@@ -179,15 +199,30 @@ const EditPlaylistPage = g(
...
+          // Reorder the playlist and refresh the lineup just in case it's
           // in the view behind the edit playlist page.
           orderPlaylist(
             metadata.playlist_id,
-            formatReorder(metadata, reorderedTracks)
+            formatReorder(playlistTrackIds, reorderedTracks)
+          )
+          // Update the playlist content track_ids so that the editPlaylist
+          // optimistically update the cached collection trackIds
+          formFields.playlist_contents.track_ids = reorderedTracks.map(
```
Comment (jowlee) [inline src/containers/edit-playlist/mobile/EditPlaylistPage.tsx:224]: I needed to do this so that line 229 `editPlaylist` would contain the correct reordered tracks, else it would incorrectly display the track order until the playlist changes are confirmed.

File: src/containers/edit-playlist/mobile/RemovePlaylistTrackDrawer.module.css [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+.title {
+  font-style: normal;
+  font-weight: var(--font-heavy);
+  font-size: 28px;
+  line-height: 34px;
+  text-align: center;
+  
+  /* Header Gradient [day] */
+  color: linear-gradient(315deg, #5B23E1 2.04%, #A22FEB 100%);
```
Comment (raymondjacobson) [inline src/containers/edit-playlist/mobile/RemovePlaylistTrackDrawer.module.css:18]: which gradient is this? shouldn't the bg image just cover this?

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/jowlee-track-playlist

File: src/containers/edit-playlist/mobile/EditPlaylistPage.tsx [hunk 3]
```diff
@@ -179,15 +199,30 @@ const EditPlaylistPage = g(
...
       }
+      // Copy the metadata playlist contents so that a reference is not changed between
+      // removing tracks, updating track order, and edit playlist
+      const playlistTrackIds = [
+        ...(metadata?.playlist_contents?.track_ids ?? [])
+      ]
+
+      for (const removedTrack of removedTracks) {
+        const { playlist_id } = metadata!
+        removeTrack(removedTrack.trackId, playlist_id, removedTrack.timestamp)
```
Comment (raymondjacobson) [inline src/containers/edit-playlist/mobile/EditPlaylistPage.tsx:210]: this remove flow needs to use UID. remove doesn't work if multiple tracks of same id in the playlist i think

**Review by raymondjacobson — COMMENTED** [review]
Screen Shot 2021-01-13 at 2 50 02 PM

Might want to not do fixed height on the drawer so we can deal w/ this

File: src/containers/edit-playlist/mobile/EditPlaylistPage.tsx (hunk 3 above)
Comment (jowlee) [reply src/containers/edit-playlist/mobile/EditPlaylistPage.tsx:210]: I think the timestamp differentiates the tracks, so it should be fine.

File: src/components/track/mobile/TrackList.tsx [hunk 4]
```diff
@@ -105,19 +109,19 @@ const TrackList = ({
...
+    const key = track?.time
+      ? `${track.trackId}:${track.time}`
+      : track.trackId.toString()
     return isReorderable ? (
-      <Draggable
-        key={track.trackId}
-        draggableId={`${track.trackId}`}
-        index={idx}
-      >
+      <Draggable key={key} draggableId={key} index={idx}>
```
Comment (jowlee) [inline src/components/track/mobile/TrackList.tsx:124]: @raymondjacobson I was going to put this in a separate PR, but this is a bug where if there are duplicate tracks in a playlist, then dragging one of them hides all of them. 
lmk what you think of this solution. I figure all playlist tracks have a unique time key which makes them unique, so I used that here.

File: src/containers/edit-playlist/mobile/EditPlaylistPage.tsx (hunk 3 above)
Comment (jowlee) [reply src/containers/edit-playlist/mobile/EditPlaylistPage.tsx:210]: lmk if it this doesn't work. I just pushed a fix for unique draggable track list item which might have been the bug you saw.

**Comment by jowlee** [comment]
> Screen Shot 2021-01-13 at 2 50 02 PM
> 
> Might want to not do fixed height on the drawer so we can deal w/ this

Just updated the styles to min height. 
Screen Shot 2021-01-14 at 9 58 55 AM

**Review by raymondjacobson — APPROVED** [review]
